.github/workflows/python-tests.yml

```diff
@@ -196,7 +202,9 @@ jobs:
         with:
           python_version: "pypy-3.8"
       - name: Run the tests
-        run: hatch -v run test:nowarn --integration_tests=true
+        env:
+          INTEGRATION_TEST: 1
+        run: hatch test -py pypy-3.8 -W default
```

Comment:
Maybe this?

```suggestion
        run: hatch test -py pypy3.8 -W default
```

Reply:
In CI can you please try showing `which -a python`?

Reply:
Done:

```
+ which -a python
/opt/hostedtoolcache/PyPy/3.8.16/x64/bin/python
/usr/bin/python
/bin/python
```

Reply:
Does `pypy3.8` work?

Reply:
Looks like that worked although it looks like there is a bug where I don't output a nice error message when there are no matching environments, I will fix but for now it should work for you.

Reply:
It is a no-op still:  https://github.com/jupyter-server/jupyter_server/actions/runs/9274819490/job/25518071400?pr=1425

Reply:
I'm so sorry for asking so many questions but can you please get some more output by using `hatch -v ...`?

Reply:
```
+ hatch -v test -py pypy3.8 -W default
No config file found, creating one with default settings now...
Success! Please see `hatch config`.
```

Reply:
I'll debug this tonight and likely add the system flag you asked for.

Reply:
Oh you need to update the matrices https://hatch.pypa.io/latest/config/internal/testing/#matrix

```toml
[[envs.hatch-test.matrix]]
python = ["3.12", "3.11", "3.10", "3.9", "3.8", "pypy3.10", "pypy3.9"]
```

Interestingly, it appears 3.8 is unsupported (perhaps I couldn't find it?) https://hatch.pypa.io/latest/plugins/environment/virtual/#pypy

I think if you add that it would still work but would be unable to download it for you should it not already exist.

So I need to definitely improve the messages tonight, thanks for debugging and let me know if this works after you update the matrices. I only do CPython by default because that's what most people want/care about.

Reply:
The matrix approach works!  https://github.com/jupyter-server/jupyter_server/actions/runs/9275245402/job/25519485224?pr=1425